odoo/addons/base/models/ir_ui_view.py

```diff
@@ -1031,6 +1010,57 @@ def postprocess_and_fields(self, node, model=None, **options):
             name_managers.extend(name_manager.children)
         return arch, models
 
+    def _postprocess_access_rights(self, tree):
+        """
+        Apply group restrictions: elements with a 'groups' attribute should
+        be removed from the view to people who are not members.
+
+        Compute and set on node access rights based on view type. Specific
+        views can add additional specific rights like creating columns for
+        many2one-based grouping views.
+        """
+        for node in tree.xpath('//*[@groups]'):
+            if not self.user_has_groups(groups=node.attrib.pop('groups')):
+                node.getparent().remove(node)
+            elif node.tag == 't' and not node.attrib:
+                # Move content of <t> blocks with no other instructions than just "groups=" to the parent
+                # and remove the <t> node.
+                # This is to keep the structure
+                # <group>
+                #   <field name="foo"/>
+                #   <field name="bar"/>
+                # <group>
+                # so the web client adds the label as expected.
+                for child in reversed(node):
+                    node.addnext(child)
+                node.getparent().remove(node)
+
+        base_model = tree.get('model_access_rights')
+        for node in tree.xpath('//*[@model_access_rights]'):
+            model = self.env[node.attrib.pop('model_access_rights')]
+            if node.tag == 'field':
+                can_create = model.check_access_rights('create', raise_exception=False)
+                can_write = model.check_access_rights('write', raise_exception=False)
+                node.set('can_create', 'true' if can_create else 'false')
+                node.set('can_write', 'true' if can_write else 'false')
+            else:
+                is_base_model = base_model == model._name
+                for action, operation in (('create', 'create'), ('delete', 'unlink'), ('edit', 'write')):
+                    if (not node.get(action) and
+                            not model.check_access_rights(operation, raise_exception=False) or
+                            not self._context.get(action, True) and is_base_model):
+                        node.set(action, 'false')
+                if node.tag == 'kanban':
```

Comment:
Maybe inverse the condition here (and also for the next if) to avoid huge lines?

(I just see that it is moved code, we can ignore)